app/models/my_academics/semesters.rb

```diff
@@ -153,22 +153,41 @@ def merge_semesters_count(data)
     end
 
     def map_enrollments(enrollment_term, term_id)
-      enrollment_term.map do |course|
-        next unless course[:role] == 'Student'
-        next if law_student? && !academic_careers.include?(course[:academicCareer])
-        mapped_course = course_info course
+      enrollment_term.map do |enrollment|
+        next unless enrollment[:role] == 'Student'
+        next if exclude_enrollment?(enrollment)
+        mapped_enrollment = course_info enrollment
         if @filtered
-          mapped_course.delete :url
+          mapped_enrollment.delete :url
         else
-          process_unfiltered_course(course, term_id)
+          process_unfiltered_enrollment(enrollment, term_id)
+        end
+        mapped_enrollment
+      end.compact
+    end
+
+    def exclude_enrollment?(enrollment)
```

Comment:
This method name should probably be the answer to the question _why are you excluding this particular enrollment?_.

If you then inverse it to use with Array#filter, the whole can read more like:

`enrollments.filter(&:has_the_thing?)`